packages/@sanity/client/sanityClient.d.ts

```diff
@@ -0,0 +1,1784 @@
+   * If no ID is specified, the currently configured ID will be returned, if any.
+   *
+   * @param id Transaction ID
+   */
+  transactionId<T extends string | undefined>(id: T): T extends string ? this : string | undefined
+
+  /**
+   * Return a plain JSON representation of the patch
+   */
+  serialize(): PatchMutationOperation
+
+  /**
+   * Return a plain JSON representation of the transaction
+   */
+  toJSON(): PatchMutationOperation
+
+  /**
+   * Commit the transaction, returning a promise that resolves to the first mutated document
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options: FirstDocumentMutationOptions): Promise<SanityDocument>
+
+  /**
+   * Commit the transaction, returning a promise that resolves to an array of the mutated documents
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options: AllDocumentsMutationOptions): Promise<SanityDocument[]>
+
+  /**
+   * Commit the transaction, returning a promise that resolves to the document ID of the first mutated document
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options: FirstDocumentIdMutationOptions): Promise<string>
+
+  /**
+   * Commit the transaction, returning a promise that resolves to the document ID of the mutated documents
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options: AllDocumentIdsMutationOptions): Promise<string[]>
+
+  /**
+   * Commit the transaction, returning a promise that resolves to the document ID of the mutated documents
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options?: BaseMutationOptions): Promise<string[]>
+
+  /**
+   * Clears the transaction of all operations
+   */
+  reset(): this
+}
+
+interface BaseClientConfig {
+  useCdn?: boolean
+  token?: string
+  apiHost?: string
+  ignoreBrowserTokenWarning?: boolean
+  withCredentials?: boolean
+
+  /**
+   * @deprecated Don't use
+   */
+  gradientMode?: boolean
+
+  /**
+   * @deprecated Don't use
+   */
+  namespace?: string
+
+  /**
+   * @deprecated Don't use
+   */
+  useProjectHostname?: boolean
+
+  /**
+   * @deprecated Don't use
+   */
+  requester?: GetItRequester
+}
+
+type ProjectlessClientConfig = BaseClientConfig & {
+  useProjectHostname: false
+  projectId?: string
+  dataset?: string
+}
+
+type ProjectClientConfig = {
+  projectId: string
+  dataset?: string
+} & BaseClientConfig
+
+type ClientConfig = ProjectClientConfig | ProjectlessClientConfig
+
+interface RequestOptions {
+  timeout?: number
+  token?: string
+}
+
+type BaseMutationOptions = RequestOptions & {
+  visibility?: 'sync' | 'async' | 'defer'
+  returnDocuments?: boolean
+  returnFirst?: boolean
+}
+
+type MutationEvent = {
+  type: 'mutation'
+  documentId: string
+  eventId: string
+  identity: string
+  mutations: Mutation[]
+  previousRev?: string
+  resultRev?: string
+  result?: SanityDocument
+  previous?: SanityDocument | null
```

Comment:
This currently is an issue with all of these typings, it's not possible to override response types.

For these, my typings had: `ListenEvent<T>` that nested into `result?: T`. I didn't have a `SanityDocument` type, but here it would simply nest into `result?: SanityDocument<T>`.

Reply:
Thanks, should be able to set type on these now!